update-checker.txt

```diff
@@ -0,0 +1,4 @@
+# List of all the files to check with the data checker
```

Comment:
There's probably no need to make this opt-in. Just grep for x-data-checker across the repo to get the list of modules that "signed up".